### PR TITLE
Tables Javadoc Improvements

### PR DESCRIPTION
**Pull request by jairmyree**
Improvements to our Javadocs for Tables

### CODE REVIEW

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/package-info.java [hunk 1]
```diff
@@ -2,6 +2,64 @@
 // Licensed under the MIT License.
 
 /**
- * Package containing the classes for Tables Clients.
+ * <p>The Azure Tables client library provides access to the database service Azure Tables. The client library supports both Azure Storage and Azure Cosmos tables. This library provides query and CRUD
```
Comment (g2vinay) [inline sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/package-info.java:5]: We can consider mentioning about service, follow that up with library below.

```java
/*
* <p>Azure Tables is a NoSQL key-value storage service offered by Microsoft Azure, which provides a highly
* scalable and cost-effective solution for storing structured data. It is a fully managed service that is designed to
* handle large volumes of structured data in a distributed environment, with low latency and high availability.
* Azure Tables stores data in tables, which are schema-less and can contain any type of data. Each table can have
* a partition key and a row key, which together form a unique primary key that can be used to retrieve individual
* records. This enables fast, efficient querying of data, especially when combined with Azure's indexing and query
* features.</p>
* 
* <p>The Azure Tables client library enables Java developers to easily interact with Azure Tables Storage Service
* from their Java applications. This library provides a set of APIs that abstract the low-level details of working with
* the Azure Tables Storage Service and allows developers to perform common operations such as creating tables,
* inserting, updating and deleting entities, querying data, and managing access control on tables and entities.
* The library supports both Azure Storage and Azure Cosmos tables. It offers both
* synchronous and asynchronous programming models. It also provides features such as retries, automatic
* pagination, and parallelism to enable efficient and reliable interactions with Azure Tables Storage Service.</p>
*/

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/package-info.java [hunk 2]
```diff
@@ -2,6 +2,64 @@
...
+ * {@link com.azure.data.tables.TableClientBuilder TableClientBuilder}.</p>
+ * 
+ * <em>* Tables created from a table client do not return a new TableClient instance. Table client instances cannot change the table they reference. To reference the newly created table, a new table
+ * client instance must be instantiated referencing the table.</em>
+ * 
+ * <p>See client builder class documentation {@link com.azure.data.tables.TableServiceClientBuilder TableServiceClientBuilder} and {@link com.azure.data.tables.TableClientBuilder TableClientBuilder}
+ * for examples of building a client.</p>
+ * <hr/>
+ * <h3>Client Operations</h3>
+ * <p>Once you have an instance of a table client or table service client, you can perform operations on the table or tables within the account. See client class documentation
```
Comment (g2vinay) [inline sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/package-info.java:50]: Add detailed samples of champion scenarios here.

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/package-info.java [hunk 3]
```diff
@@ -2,6 +2,64 @@
...
+ * </ul>
+ * <hr/>
+ * <h3>Authenticate the Client</h3>
+ * <p>In order to build a valid table client or table service client, you will need to authenticate the client using an accepted method of authentication. The supported forms of authentication are:
+ * <ul>
+ * <li>Connection String</li>
+ * <li>Shared Key</li>
+ * <li>Shared Access Signature (SAS)</li>
+ * <li>Token Credential</li>
+ * </ul>
```
Comment (g2vinay) [inline sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/package-info.java:25]: link out to MS docs for user to read more about these auth types.

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/package-info.java [hunk 4]
```diff
@@ -2,6 +2,64 @@
...
+ *  service client.</em>
+ * <p>See client builder class documentation {@link com.azure.data.tables.TableServiceClientBuilder TableServiceClientBuilder} and {@link com.azure.data.tables.TableClientBuilder TableClientBuilder}
+ * for examples of authenticating a client.</p>
+ * <hr/>
+ * <h3>Building a Client</h3>
+ * <p>The {@link com.azure.data.tables.TableServiceClient TableServiceClient} and {@link com.azure.data.tables.TableServiceAsyncClient TableServiceAsyncClient} provide access to the tables within an 
+ * Azure Storage or Azure Cosmos account. A table service client can create, list, and delete tables. It also provides access to a table client that can be used to perform CRUD operations on entities
+ * within a table. You can instantiate a table service client using an instance of {@link com.azure.data.tables.TableServiceClientBuilder TableServiceClientBuilder TableServiceClientBuilder}.</p>
+ * 
+ * insert CodeSnippet: TablesJavaInstantiateServiceClient
```
Comment (g2vinay) [inline sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/package-info.java:36]: When adding samples, describe in text above what the sample is doing as well.

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableServiceClientBuilder.java [hunk 5]
```diff
@@ -39,35 +39,48 @@
...
+ * <h3>Initializing a Service Client</h3>
+ * 
+ * <p>The minimal configuration options required by {@link TableServiceClientBuilder} to build a {@link TableServiceClient} or
+ * {@link TableServiceAsyncClient} are an {@link String endpoint} and a form of authentication,
+ * which can be set via: {@link TableServiceClientBuilder#connectionString(String)},
+ * {@link TableServiceClientBuilder#credential(AzureNamedKeyCredential)}, {@link TableServiceClientBuilder#credential(TokenCredential)}, {@link TableServiceClientBuilder#credential(AzureSasCredential)},
+ *  or {@link TableServiceClientBuilder#sasToken(String)}</p>
  *
- * <p><strong>Samples to construct a sync client</strong></p>
- * <!-- src_embed com.azure.data.tables.tableServiceClient.instantiation -->
```
Comment (g2vinay) [inline sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableServiceClientBuilder.java:54]: These can still be retained here, I think.

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableServiceClient.java [hunk 6]
```diff
@@ -49,24 +49,19 @@
...
- * <pre>
- * TableServiceClient tableServiceClient = new TableServiceClientBuilder&#40;&#41;
- *     .endpoint&#40;&quot;https:&#47;&#47;myvault.azure.net&#47;&quot;&#41;
- *     .credential&#40;new AzureNamedKeyCredential&#40;&quot;name&quot;, &quot;key&quot;&#41;&#41;
- *     .buildClient&#40;&#41;;
- * </pre>
- * <!-- end com.azure.data.tables.tableServiceClient.instantiation -->
+ * <p>See {@link TableServiceClientBuilder} documentation for more information on constructing a client.</p>
+ * <h4>Usage Code Samples</h4>
  *
```
Comment (g2vinay) [inline sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableServiceClient.java:65]: We should add champion scenarios code samples here.

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableServiceAsyncClient.java [hunk 7]
```diff
@@ -48,24 +48,19 @@
...
- * <pre>
- * TableServiceAsyncClient tableServiceAsyncClient = new TableServiceClientBuilder&#40;&#41;
- *     .endpoint&#40;&quot;https:&#47;&#47;myvault.azure.net&#47;&quot;&#41;
- *     .credential&#40;new AzureNamedKeyCredential&#40;&quot;name&quot;, &quot;key&quot;&#41;&#41;
- *     .buildAsyncClient&#40;&#41;;
- * </pre>
- * <!-- end com.azure.data.tables.tableServiceAsyncClient.instantiation -->
+ * <p>See {@link TableServiceClientBuilder} documentation for more information on constructing a client.</p>
+ * <h3>Usage Code Samples</h3>
  *
```
Comment (g2vinay) [inline sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableServiceAsyncClient.java:64]: we should add champion scenarios code samples here.

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableAsyncClient.java [hunk 8]
```diff
@@ -73,27 +73,25 @@
...
  * <p>The client encapsulates the URL for the table within the Tables service endpoint, the name of the table, and the
  * credentials for accessing the storage or CosmosDB table API account. It provides methods to create and delete the
  * table itself, as well as methods to create, upsert, update, delete, list, and get entities within the table. These
  * methods invoke REST API operations to make the requests and obtain the results that are returned.</p>
  *
+ * <h2>Creating an Asynchronous Table Client</h2>
  * <p>Instances of this client are obtained by calling the {@link TableClientBuilder#buildAsyncClient()} method on a
  * {@link TableClientBuilder} object.</p>
- *
- * <p><strong>Samples to construct an async client</strong></p>
```
Comment (g2vinay) [inline sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableAsyncClient.java:86]: This sample can be retained here.

**Comment by jairmyree** [comment]
In progress

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/package-info.java [hunk 9]
```diff
@@ -2,6 +2,257 @@
...
+ * <pre>
+ * tableServiceClient.deleteTable&#40;&quot;tableName&quot;&#41;;
+ * </pre>
+ * <!-- end com.azure.data.tables.TableServiceClient.deleteTable.package#String -->
+ *
+ * <p><strong>Note:</strong> For asynchronous sample, refer to {@link com.azure.data.tables.TableServiceAsyncClient TableServiceAsyncClient}</p>
+ *
+ * <hr/>
+ *
+ * <h3>Retrieve a {@link com.azure.data.tables.TableClient TableClient} from a {@link com.azure.data.tables.TableServiceClient TableServiceClient}</h3>
```
Comment (g2vinay) [inline sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/package-info.java:146]: Add an H2 for Table Client Usage/Examples will help to indicate the start of new series of samples here.

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableAsyncClient.java [hunk 10]
```diff
@@ -75,26 +75,203 @@
...
+ *     .addProperty&#40;&quot;Property&quot;, &quot;Value&quot;&#41;;
+ *
+ * tableAsyncClient.createEntity&#40;tableEntity&#41;
+ *     .contextWrite&#40;Context.of&#40;&quot;key1&quot;, &quot;value1&quot;, &quot;key2&quot;, &quot;value2&quot;&#41;&#41;
+ *     .subscribe&#40;unused -&gt;
+ *         System.out.printf&#40;&quot;Table entity with partition key '%s' and row key '%s' was created.&quot;, &quot;partitionKey&quot;,
+ *             &quot;rowKey&quot;&#41;&#41;;
+ * </pre>
+ * <!-- end com.azure.data.tables.tableAsyncClient.createEntity#TableEntity -->
+ *
```
Comment (g2vinay) [inline sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableAsyncClient.java:124]: Add here and for other samples,
 * <p><strong>Note:</strong> for asynchronous sample, refer to {@link async client}</p>

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableClient.java [hunk 11]
```diff
@@ -78,26 +78,199 @@
...
  *     .tableName&#40;&quot;myTable&quot;&#41;
  *     .buildClient&#40;&#41;;
  * </pre>
- * <!-- end com.azure.data.tables.tableClient.instantiation -->
+ * <!-- end com.azure.data.tables.tableClient.connectionstring.instantiation -->
+ *
+ * <p>For more information on building and authenticating, see the {@link TableClientBuilder} documentation.</p>
+ *
+ * <p>The following code samples provide examples of common operations preformed with this client.</p>
+ *
```
Comment (g2vinay) [inline sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableClient.java:106]: Add here and for other samples,
 * <p><strong>Note:</strong> for asynchronous sample, refer to {@link async client}</p>

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableServiceAsyncClient.java [hunk 12]
```diff
@@ -48,26 +48,173 @@
...
+ *
+ * <!-- src_embed com.azure.data.tables.tableServiceAsyncClient.createTable#String -->
+ * <pre>
+ * tableServiceAsyncClient.createTable&#40;&quot;myTable&quot;&#41;
+ *     .contextWrite&#40;Context.of&#40;&quot;key1&quot;, &quot;value1&quot;, &quot;key2&quot;, &quot;value2&quot;&#41;&#41;
+ *     .subscribe&#40;tableAsyncClient -&gt;
+ *         System.out.printf&#40;&quot;Table with name '%s' was created.&quot;, tableAsyncClient.getTableName&#40;&#41;&#41;&#41;;
+ * </pre>
+ * <!-- end com.azure.data.tables.tableServiceAsyncClient.createTable#String -->
+ *
```
Comment (g2vinay) [inline sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableServiceAsyncClient.java:94]: Add here and for other samples,
 * <p><strong>Note:</strong> for synchronous sample, refer to {@link sync client}</p>

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableServiceClient.java [hunk 13]
```diff
@@ -49,26 +49,180 @@
...
+ * <p>The following sample deletes the table with the name "myTable".</p>
+ *
+ * <!-- src_embed com.azure.data.tables.tableServiceClient.deleteTable#String -->
+ * <pre>
+ * tableServiceClient.deleteTable&#40;&quot;myTable&quot;&#41;;
+ *
+ * System.out.printf&#40;&quot;Table with name '%s' was deleted.&quot;, &quot;myTable&quot;&#41;;
+ * </pre>
+ * <!-- end com.azure.data.tables.tableServiceClient.deleteTable#String -->
+ *
```
Comment (g2vinay) [inline sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableServiceClient.java:112]: Add here and for other samples,
 * <p><strong>Note:</strong> for asynchronous sample, refer to {@link async client}</p>

**Review by g2vinay — APPROVED** [review]

**Pull request merged**